src/fluree/db/flake/index/novelty.cljc

```diff
@@ -98,47 +98,118 @@
 
 (defn rebalance-children
   [branch t child-nodes]
-  (let [target-count (/ *overflow-children* 2)]
-    (->> child-nodes
-         (partition-all target-count)
-         (map (fn [kids]
-                (reconstruct-branch branch t kids)))
-         update-sibling-leftmost)))
+  (let [median-i       (-> child-nodes count (quot 2))
+        left-children  (subvec (vec child-nodes) 0 median-i)
+        right-children (subvec (vec child-nodes) median-i)
+        left-branch    (reconstruct-branch branch t left-children)
+        right-branch   (reconstruct-branch branch t right-children)]
+    (update-sibling-leftmost [left-branch right-branch])))
+
+(defn calculate-split-params
+  "Calculates parameters needed for splitting a leaf into N leaves.
+  Returns a map with:
+  - :flake-sizes - vector of [flake size] tuples
+  - :total-bytes - total size of all flakes
+  - :target-leaves - number of leaves to create
+  - :bytes-per-leaf - target bytes per leaf"
+  [flakes]
+  (let [flake-vec             (vec flakes)
```

Comment:
Why is this necessary? I think `flakes` is a sorted set, which is also `Seqable`.